tc/core/cuda/cuda_tc_executor_new_api.cc

```diff
@@ -0,0 +1,163 @@
+namespace {
+// Append ordered values to the kernel name, separated by "_".
+std::string specializeKernelName(
+    const std::string& tcName,
+    std::vector<int> params) {
+  std::stringstream ss;
+  ss << tcName;
+  for (auto i : params) {
+    ss << "_" << i;
+  }
+  return ss.str();
+}
+
+std::vector<int> narrowParamsVector(const std::vector<long>& params) {
+  std::vector<int> result;
+  result.reserve(params.size());
+  for (auto l : params) {
+    CHECK_GE(l, std::numeric_limits<int>::min()) << "parameter underflows int";
+    CHECK_LE(l, std::numeric_limits<int>::max()) << "parameter overflows int";
+    result.push_back(static_cast<int>(l));
+  }
+  return result;
+}
+} // namespace
+
+CudaTcExecutor::CudaTcExecutor(
+    const std::vector<TensorInfo>& inputsInfo,
+    const std::vector<TensorInfo>& outputsInfo,
+    const tc2halide::HalideComponents& halideComponents,
+    const typename CudaBackend::CompilationResultType& compilationResult)
+    : TcExecutor<CudaBackend>(
+          inputsInfo,
+          outputsInfo,
+          halideComponents,
+          compilationResult) {
+  auto t0 = std::chrono::high_resolution_clock::now();
+  // force unloading in case we JIT with the same name/input/outputs with
+  // different options.
+  this->clearRuntimeCompiledFunction();
+  rtcFun_ = CudaRTCFunction::Compile(
+      compilationResult.specializedName, compilationResult.source);
+  grid_ = compilationResult.grid;
+  block_ = compilationResult.block;
+  auto t1 = std::chrono::high_resolution_clock::now();
+  LOG_IF(INFO, FLAGS_debug_tc_mapper)
+      << "[COMPILE] Compiling with host JIT compiler took: "
+      << std::chrono::duration_cast<std::chrono::milliseconds>(t1 - t0).count()
+      << "ms" << std::endl;
+}
+
+CudaCompilationResult CudaBackend::compileWithTcMapper(
+    const std::string& tcName,
+    tc2halide::HalideComponents halideComponents,
+    const std::vector<const DLConstTensor*>& inputs,
+    /* TODO: in the future also pass outputs for stride and alignment info */
+    const CudaMappingOptions& options) {
+  // A bit chicken-and-eggy, need scop from TC to have the space to build the
+  // context to specialize the scop..
+  auto scopTmp = polyhedral::Scop::makeScop(
+      isl::with_exceptions::globalIslCtx(), halideComponents);
+  auto globalParameterContext = scopTmp->makeContextFromInputs(inputs);
+  scopTmp = polyhedral::Scop::makeSpecializedScop(
+      *scopTmp,
+      globalParameterContext.intersect(scopTmp->globalParameterContext));
+  LOG_IF(INFO, FLAGS_debug_tc_mapper) << options;
+  LOG_IF(INFO, FLAGS_debug_tc_mapper) << *(scopTmp->scheduleRoot());
+
+  // Now we can build stuff
+  auto mappedScop =
+      polyhedral::MappedScop::makeWithOuterBlockInnerThreadStrategy(
+          std::move(scopTmp), options);
+  LOG_IF(INFO, FLAGS_debug_tc_mapper) << "Mapped schedule:" << std::endl
+                                      << *(mappedScop->schedule());
+
+  auto parameters = narrowParamsVector(
```

Comment:
the call to narrowParamsVector (and the function itself since it has only once call place) is only necessary because specializeKernelName takes a vector of ints with the parameter values.  I don't see why, it looks like it was needed to silence a warning.  Let's make specializeKernelName accept longs, or a template argument type, and remove the narrowing completely.